src/main/java/core/basesyntax/file/DataConverterImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.file;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputReport) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
```

Comment:
Consider adding validation to check if the `parts` array has the expected number of elements before accessing them. This will prevent `ArrayIndexOutOfBoundsException` if the input data is malformed.